o3de_package_scripts/build_all_packages.py

```diff
@@ -0,0 +1,148 @@
+            print(f"    - {package_name} already at {found}.")
+            packages_already_found_on_server.append(package_name)
+        
+        if not package_name in packages_already_found_on_server:
+            package_abspath = folder_packages[package_name]
+            package_info_file_path = os.path.join(package_abspath, CommonUtils.package_descriptor_name)
+            # over here we'd sync the folder, if necessary, using p4 or git or whatever.
+            # for now we assume its all fetched.
+            if not os.path.exists(package_info_file_path):
+                print(f"Error: Package info file not found: {package_info_file_path} ... skipping")
+                failed_folder_packages.append(package_name)
+                exitCode = 1
+                continue
+            try:
+                data = CommonUtils.ReadPackageInfo(package_info_file_path)
+                if data['PackageName'] != package_name:
+                    raise KeyError(f"Package {package_name} has a PackageInfo.json that claims its {data['PackageName']} instead.")
+            
+                # build it:
+                if not PackageUpFolder(package_abspath, output_folder):
+                    print(f"Error:  {package_name} failed to package up correctly.")
+                    exitCode = 1
+                    failed_folder_packages.append(package_name)
+
+            except Exception as e:
+                print(f"Error:  {package_name} {e}") 
+                exitCode = 1
+                failed_folder_packages.append(package_name)
+                traceback.print_exc()
+                continue
+
+    if packages_already_found_on_server:
+        print("The following packages were skipped as they were already on the server(s)")
+        for package_name in packages_already_found_on_server:
+            print(f"   [SKIPPED] - {package_name}")
+    
+    if failed_source_packages:
+        print("WARNING: These packages failed to build from source:")
+        for package_name in failed_source_packages:
+            print(f"   [FAILED] - {package_name}")
+
+    if failed_folder_packages:
+        print("WARNING: These packages failed to build from folder:")
+        for package_name in failed_folder_packages:
+            print(f"   [FAILED] - {package_name}")
+    
+    return exitCode
+
+if __name__ == "__main__":
+    parser = argparse.ArgumentParser(description='Builds any missing packages into the packages folder, based on package list json files.')
+    CommonUtils.AddCommonArgs(parser)
+    FindPackageUtils.AddServerArgs(parser)
+    args = parser.parse_args()
+    CommonUtils.PostArgParse(args)
+
+    if args.profile_name:
+        print(f"Using AWS profile: {args.profile_name}")
+
+    if not args.server_urls:
+        print("You must specify server urls to check in order to use this script")
+        print("Either set LY_SERVER_URLS (semi colon list) or specify it in the command line in --server_urls (semi colon list)")
```

Comment:
Should that be `LY_PACKAGE_SERVER_URLS` instead of `LY_SERVER_URLS`?

Reply:
Yeas!  I'll change it.  Thanks for spotting.